rust/worker/src/execution/orchestration/knn.rs

```diff
@@ -0,0 +1,695 @@
+use chroma_blockstore::provider::BlockfileProvider;
+use chroma_distance::{DistanceFunction, DistanceFunctionError};
+use chroma_error::{ChromaError, ErrorCodes};
+use chroma_index::hnsw_provider::HnswIndexProvider;
+use chroma_types::MetadataValue;
+use thiserror::Error;
+use tokio::sync::oneshot::{self, error::RecvError, Sender};
+use tonic::async_trait;
+use tracing::Span;
+
+use crate::{
+    execution::{
+        dispatcher::Dispatcher,
+        operator::{wrap, TaskError, TaskResult},
+        operators::{
+            fetch_log::{FetchLogError, FetchLogOperator, FetchLogOutput},
+            fetch_segment::{FetchSegmentError, FetchSegmentOperator, FetchSegmentOutput},
+            filter::{FilterError, FilterInput, FilterOperator, FilterOutput},
+            knn::{KnnOperator, RecordDistance},
+            knn_hnsw::{KnnHnswError, KnnHnswInput, KnnHnswOutput},
+            knn_log::{KnnLogError, KnnLogInput, KnnLogOutput},
+            knn_merge::{KnnMergeError, KnnMergeInput, KnnMergeOperator, KnnMergeOutput},
+            knn_projection::{
+                KnnProjectionError, KnnProjectionInput, KnnProjectionOperator, KnnProjectionOutput,
+            },
+        },
+        orchestration::common::terminate_with_error,
+    },
+    system::{ChannelError, Component, ComponentContext, ComponentHandle, Handler, System},
+};
+
+/// The `knn` module contains two orchestrator: `KnnFilterOrchestrator` and `KnnOrchestrator`.
```

Comment:
Feels like this should be one orchestrator or the filter orchestrator should be able to be reused with get()?

Reply:
It is possible to divide `GetOrchestrator` into two parts, where the first part is identical to the `KnnFilterOrchestrator` here.

Reply:
Reminder for myself: Maybe consider reusing `KnnFilterOrchestrator` for `GetOrchestrator`